paddle/fluid/pybind/imperative.cc

```diff
@@ -220,6 +221,80 @@ void BindImperative(py::module *m_ptr) {
 
   BindOpFunctions(&m);
 
+#ifndef _WIN32
+  // Dygraph DataLoader signal handler
+  m.def("_set_process_pid", [](int64_t key, pid_t pid) {
+    imperative::SetLoadProcessPID(key, pid);
+  });
+  m.def("_erase_process_pid",
+        [](int64_t key) { imperative::EraseLoadProcessPID(key); });
+  m.def("_set_process_signal_handler",
+        []() { imperative::SetLoadProcessSignalHandler(); });
+  m.def("_throw_error_if_process_failed",
+        []() { imperative::ThrowErrorIfLoadProcessFailed(); });
+
+  // Dygraph DataLoader reader process & thread related functions
+  m.def("_convert_to_tensor_list",
+        [](py::object &obj) -> py::list {
+          // 0. input data check
+          PADDLE_ENFORCE(
+              py::isinstance<py::tuple>(obj) || py::isinstance<py::list>(obj),
+              platform::errors::InvalidArgument(
+                  "The batch data read into DataLoader is illegal."
+                  "Expected data type is tuple or list, but received %s",
+                  obj.get_type()));
+          py::list batch = py::cast<py::list>(obj);
+          py::list tensors;
+          for (size_t i = 0; i < batch.size(); ++i) {
+            // 1. cast to python array
+            auto array = batch[i].cast<py::array>();
+            PADDLE_ENFORCE_NE(
+                string::Sprintf("%s", array.dtype()).compare("object"), 0,
+                platform::errors::InvalidArgument(
+                    "Faild to convert input data to a regular ndarray.\n  * "
+                    "Usually this means the input data contains nested "
+                    "lists with different lengths.\n  * Check the reader "
+                    "function passed to 'set_(sample/sample_list/batch)"
+                    "_generator' to locate the data causes this issue."));
+            // 2. construcct LoDTensor
+            framework::LoDTensor t;
+            SetTensorFromPyArray<platform::CPUPlace>(
+                &t, array, platform::CPUPlace(), true);
+            // 3. allocate shared memory
+            memory::allocation::MemoryMapAllocator allocator;
+            void *data_ptr = reinterpret_cast<void *>(
+                reinterpret_cast<uintptr_t>(t.Holder()->ptr()) + t.offset());
+            size_t data_size = t.memory_size();
```

Comment:
`memory_size()` may be larger than `t.numel() * SizeOfType(t.type())`. How about use `t.numel() * SizeOfType(t.type())` instead?

Reply:
done